BLE_GattServer_AddService/source/main.cpp

```diff
@@ -30,48 +31,52 @@ class HeartrateDemo : ble::Gap::EventHandler {
     HeartrateDemo(BLE &ble, events::EventQueue &event_queue) :
         _ble(ble),
         _event_queue(event_queue),
-        _led1(LED1, 1),
-        _connected(false),
-        _hr_uuid(GattService::UUID_HEART_RATE_SERVICE),
-        _hr_counter(100),
-        _hr_service(ble, _hr_counter, HeartRateService::LOCATION_FINGER),
-        _adv_data_builder(_adv_buffer) { }
-
-    void start() {
-        _ble.gap().setEventHandler(this);
+        _heartrate_uuid(GattService::UUID_HEART_RATE_SERVICE),
+        _heartrate_value(100),
+        _heartrate_service(ble, _heartrate_value, HeartRateService::LOCATION_FINGER),
+        _adv_data_builder(_adv_buffer)
+    {
+    }
 
+    void start()
+    {
         _ble.init(this, &HeartrateDemo::on_init_complete);
 
-        _event_queue.call_every(500, this, &HeartrateDemo::blink);
-        _event_queue.call_every(1000, this, &HeartrateDemo::update_sensor_value);
-
         _event_queue.dispatch_forever();
     }
 
 private:
     /** Callback triggered when the ble initialization process has finished */
-    void on_init_complete(BLE::InitializationCompleteCallbackContext *params) {
+    void on_init_complete(BLE::InitializationCompleteCallbackContext *params)
+    {
         if (params->error != BLE_ERROR_NONE) {
             printf("Ble initialization failed.");
             return;
         }
 
         print_mac_address();
 
+        /* this allows us to receive events like onConnectionComplete() */
+        _ble.gap().setEventHandler(this);
+
+        /* heart rate value updated every second */
+        _event_queue.call_every(1000ms, this, &HeartrateDemo::update_sensor_value);
```

Comment:
what is the point of this lambda? it just calls the same function the call_every() calls. Do you mean I put the body of the update_sensor_value() inside the lambda?

Reply:
The lambda would be just what I've written, there's too many ways of writing callbacks in my opinion, the `this` pointer can come first or be put after the member function pointer, it will change if multiple parameters are used, ...
With lambdas a single syntax can be used.

Reply:
I agree, it is not the case where it shines the most but as seen in the periodic advertising example, when expressions became more complex, a lambda is more readable.